### DISCUSSION TITLE
컨디셔널한 렌더링이 최선일까.....에 대한 고민

### DISCUSSION THREAD

**Discussion by harry21-kr** [munyeol-Yoon/Permu#45] [자유게시판]
`useAuth` 훅을 통해 불러오는 user의 정보는 초기에 undefined를 제공합니다.

따라서 요 친구를 사용하는 훅은 정보를 받아온 후 다시 실행이 되어야해 결과적으로 초기에는 `undefined` 값을 반환해
렌더링하는 코드에서는 컨디셔널하게 렌더링을 하여야 합니다.

차 후 이 부분이 개선되어야 조금 더 매끄러운 UX가 완성될 것 같습니다.

예를 들어, 서버에서 로그인한 유저의 정보를 미리 받아와 `loggedUser` 에 넣어준다던지 등등 ...

**Comment by munyeol-Yoon** [comment]
이부분을 해결하려면 결국 서버쪽 백엔드 코드 작업이 불가피해집니다.
OAuth 를 메인 로그인으로 사용하는 지금 프로덕트에는 스코프가 급격하게 증가해 기간동안은 어려울 것으로 예상 됩니다.
백엔드가 구축될때 반영하도록 하면 좋을 것 같아요!
